scripts/generate_resource/main.go

```diff
@@ -0,0 +1,102 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"regexp"
+	"strings"
+	"text/template"
+
+	"github.com/gertd/go-pluralize"
+	"github.com/iancoleman/strcase"
+)
+
+var validResourceName = regexp.MustCompile(`^[a-zA-Z_]+$`).MatchString
```

Comment:
There is a difference in the validation between invoking the command from the Makefile and invoking with go.

When I use the make file I get this error:

```
[awinkler@awinkler-NJ1XWDG2N0 go-tfe (resource-scaffolding-v2 ✗)]$ make generate resource=✨
Makefile:13: *** Undefined RESOURCE.  Stop.
```

But then if I cd into `scripts/generate_resource` and run the following, I get a more helpful error message:

```
4156 scripts/generate_resource git:(resource-scaffolding-v2) ✗ » go run . ✨
2022/09/26 14:48:10 resource name can only contain letters or underscores.
exit status 1
```

It would be nice if the error from the makefile invocation was the same as the go version.

Reply:
Ah! I tested this again:

```
[awinkler@awinkler-NJ1XWDG2N0 go-tfe (resource-scaffolding-v2 ✗)]$ RESOURCE=✨ make generate
2022/09/26 14:59:32 resource name can only contain letters or underscores.
exit status 1
make: *** [generate] Error 1
```

And the error is the same. I think I was using the wrong format of the command.